TailLight/SetTaillightBlack.cpp

```diff
@@ -0,0 +1,268 @@
+#include "driver.h"
+#include <Hidport.h>
+
+#include "SetBlack.h"
+
+EVT_WDF_REQUEST_COMPLETION_ROUTINE  SetBlackCompletionRoutine;
+EVT_WDF_WORKITEM                    SetBlackWorkItem;
+
+NTSTATUS CreateWorkItemForIoTargetOpenDevice(WDFDEVICE device, 
+    CONST UNICODE_STRING& symLink)
+    /*++
+
+    Routine Description:
+
+        Creates a WDF workitem to do the SetBlack() call after the driver
+        stack has initialized.
+
+    Arguments:
+
+        Device - Handle to a pre-allocated WDF work item.
+
+    Requirements:
+        Must be synchronized to the device.
+
+    --*/
+{    
+    WDFWORKITEM hWorkItem = 0;
+    NTSTATUS status = STATUS_PNP_DRIVER_CONFIGURATION_INCOMPLETE;
+    {
+        WDF_WORKITEM_CONFIG        workItemConfig;
+        WDF_OBJECT_ATTRIBUTES      workItemAttributes;
+        WDF_OBJECT_ATTRIBUTES_INIT(&workItemAttributes);
+        WDF_OBJECT_ATTRIBUTES_INIT_CONTEXT_TYPE(&workItemAttributes,
+            SET_BLACK_WORKITEM_CONTEXT);
+        workItemAttributes.ParentObject = device;
+
+        DEVICE_CONTEXT* pDeviceContext = WdfObjectGet_DEVICE_CONTEXT(device);
+
+        // It's possible to get called twice.
```

Comment:
Please document the situation that could trigger this function to be called twice for a given device?

Reply:
Thanks for adding a documentation link to this claim.